Functions/Import/xASL_imp_CheckImportSettings.m

```diff
@@ -64,7 +64,7 @@
 		nRightBrackets = nRightBrackets + numel(rightBracket);
 	end
 
-	nTokens = sum(x.modules.import.imPar.tokenOrdering>0);
+	nTokens = max(x.modules.import.imPar.tokenOrdering);
```

Comment:
This is incorrect, if I'm correct ;)

`tokenOrdering` is something like [1 0 5 3] (in the example that subject is the first folder level, there is no visit, the session is the 5th folder level (with some dummy folders in between that we want to keep as we want to change the `sourcedata` as minimally as possible), and scan as third folder layer) so for the number of tokens you need the sum of non-zero tokens. Note that the zeros should be appended before this part if they were omitted by the user (e.g., [1 5 3]

Reply:
No you're not correct :)
sum([1 0 5 3]>0) will return 3. But we need tokens 1, 5, and 3. So we need 5 tokens defined so that we skip 2 and 4. 

We use 3 tokens, but tokens 1 3 5, so maximum token number 5. So we need 5 () defined in folderhierarchy, not 3.

Reply:
I would reserve tokens for those that have a meaning, they are a "token" for something. () without a meaning is not a token, but denotes a capturing group.

So () == brackets
tokens == defined in the aliases
capturing groups == all brackets (can be a group or a group+token)